components/tools/OmeroPy/test/integration/gatewaytest/test_connection.py

```diff
@@ -77,6 +78,42 @@ def g2_getTestImage():
         gatewaywrapper._has_connected = False
         gatewaywrapper.doDisconnect()
 
+    def testTerminateAllClients(self, gatewaywrapper, author_testimg):
+        # author_testimg in args to make sure the image has been imported
+        gatewaywrapper.loginAsAuthor()
+        assert gatewaywrapper.getTestImage() is not None
+        gatewaywrapper.gateway.terminateAllClients()
+        pytest.raises(Ice.ConnectionLostException, gatewaywrapper.getTestImage)
+        gatewaywrapper._has_connected = False
+        gatewaywrapper.doDisconnect()
+        gatewaywrapper.loginAsAuthor()
+        assert gatewaywrapper.getTestImage() is not None
+        gatewaywrapper.gateway.terminateAllClients(softclose=False)
+        pytest.raises(Ice.ConnectionLostException, gatewaywrapper.getTestImage)
+        gatewaywrapper._has_connected = False
+        gatewaywrapper.doDisconnect()
+        # Also make sure softclose does the right thing
+        gatewaywrapper.loginAsAuthor()
+        g2 = gatewaywrapper.gateway.clone()
+
+        def g2_getTestImage():
+            return dbhelpers.getImage(g2, 'testimg1')
+        assert g2.connect(gatewaywrapper.gateway._sessionUuid)
+        assert gatewaywrapper.getTestImage() is not None
+        assert g2_getTestImage() is not None
+        g2.terminateAllClients(softclose=True)
+        pytest.raises(Ice.ConnectionLostException, g2_getTestImage)
+        assert gatewaywrapper.getTestImage() is not None
+        g2 = gatewaywrapper.gateway.clone()
+        assert g2.connect(gatewaywrapper.gateway._sessionUuid)
+        assert gatewaywrapper.getTestImage() is not None
+        assert g2_getTestImage() is not None
+        g2.terminateAllClients(softclose=False)
+        pytest.raises(Ice.ConnectionLostException, g2_getTestImage)
+        pytest.raises(Ice.ObjectNotExistException, gatewaywrapper.getTestImage)
```

Comment:
Not sure I understand why this is an ObjectNotExistException but everywhere else we get an Ice. ConnectionLostException (including for the same call above)?